### ISSUE TITLE
Accelerate core functions with SIMD intrinsics

### ISSUE THREAD

**Issue by chawyehsu** [albertobsd/keyhunt#36]
Some core computation functions like sha256, bloom (and some others maybe) are implemented as per standards. 

To gain more computation speed, SIMD instructions accelerated implementations of these functions could help. For instance, we know a [SIMD accelerated version of sha256](https://github.com/bitcoin/bitcoin/blob/master/src/crypto/sha256.cpp) is used by the bitcoin-core client.

_We Need to Go Deeper._

**Comment by albertobsd** [comment]
> _We Need to Go Deeper._

Yes you are right, just now im implementing the Jean Luc Pons `SECP256K1` [librery](https://github.com/JeanLucPons/BSGS/tree/master/SECPK1), that work in very similar way than the libgmp, but it use some group Modulo Operations with SSE2 Operations to boost his speed to create publickeys up to 5 times faster. This will boost all the speed from all modes from address to bsgs mode.

Once that i finish that part i will try to implement your suggestion with that version of sha256.

Best regards

**Comment by MajMcCloud** [comment]
> Excellent, thanks for sharing, to check if it is working welll you can run this test:
> 
> ```
> ./keyhunt -m rmd160 -f tests/1to32.rmd -r 1:FFFFFFFF -l compress
> ```
> 
> The will check if the rmd output is OK or not, is in the readme https://github.com/albertobsd/keyhunt#rmd160-mode
> 
> Regards!

Sadly i can't test it, cause it seems there are other issues as well. So after changing what I have stated above, there are rising multiple different errors, so no easy way for now to fix them.
